### PR TITLE
Add changes to archive favorites service methods

### PR DESCRIPTION
**Pull request by scott062**
- Proposed changes to dev1 
- Add `archived=False` to a bunch of other class views in AP/PV

### CODE REVIEW

File: talentmap_api/fsbid/services/available_positions.py [hunk 1]
```diff
@@ -388,20 +388,21 @@ def archive_favorites(aps, request, favoritesLimit=FAVORITES_LIMIT):
...
+        returned_ids = get_ap_favorite_ids(QueryDict(f"id={pos_nums}&limit=999999&page=1"), request.META['HTTP_JWT'], f"{request.scheme}://{request.get_host()}")
         # Need to determine which ids need to be archived using comparison of lists above
         outdated_ids = []
-        for fav_id in list_favs:
-            if fav_id in returned_ids:
-                outdated_ids.append(fav_id)
+        if isinstance(returned_ids, list):
+            for fav_id in list_favs:
+                if fav_id not in returned_ids:
+                    outdated_ids.append(fav_id)
```
Comment (scott062) [inline talentmap_api/fsbid/services/available_positions.py:397]: Note the flipped logic `not in` vs `in`

File: talentmap_api/fsbid/services/projected_vacancies.py [hunk 2]
```diff
@@ -268,9 +268,10 @@ def archive_favorites(pvs, request, favoritesLimit=FAVORITES_LIMIT):
         returned_ids = get_pv_favorite_ids(QueryDict(f"id={pos_nums}&limit=999999&page=1"), request.META['HTTP_JWT'], f"{request.scheme}://{request.get_host()}")
         # Need to determine which ids need to be archived using comparison of lists above
         outdated_ids = []
-        for fav_id in list_favs:
-            if fav_id not in returned_ids:
-                outdated_ids.append(fav_id)
+        if isinstance(returned_ids, list):
```
Comment (mjoyce91) [inline talentmap_api/fsbid/services/projected_vacancies.py:271]: 👍

File: talentmap_api/fsbid/services/available_positions.py [hunk 3]
```diff
@@ -388,20 +388,21 @@ def archive_favorites(aps, request, favoritesLimit=FAVORITES_LIMIT):
         # List favs is list of integers instead of strings for comparison
         list_favs = list(map(lambda x: int(x), aps))
         # Ids from fsbid that have filled position code (double checks they are invalid)
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/available_positions.py:390]: i think this comment is left over from when the param '&cps_codes=FP' was being used?

File: talentmap_api/fsbid/services/available_positions.py [hunk 4]
```diff
@@ -387,21 +387,22 @@ def archive_favorites(aps, request, favoritesLimit=FAVORITES_LIMIT):
...
-            if fav_id in returned_ids:
-                outdated_ids.append(fav_id)
-        if len(outdated_ids) > 0:
-            AvailablePositionFavorite.objects.filter(cp_id__in=outdated_ids).update(archived=True)
+        if isinstance(returned_ids, list):
+            for fav_id in list_favs:
+                if fav_id not in returned_ids:
+                    outdated_ids.append(fav_id)
+            if len(outdated_ids) > 0:
+                AvailablePositionFavorite.objects.filter(cp_id__in=outdated_ids).update(archived=True)
```
Comment (scott062) [inline talentmap_api/fsbid/services/available_positions.py:399]: Indent

File: talentmap_api/fsbid/services/projected_vacancies.py [hunk 5]
```diff
@@ -268,11 +268,12 @@ def archive_favorites(pvs, request, favoritesLimit=FAVORITES_LIMIT):
...
-            if fav_id not in returned_ids:
-                outdated_ids.append(fav_id)
-        if len(outdated_ids) > 0:
-            ProjectedVacancyFavorite.objects.filter(fv_seq_num__in=outdated_ids).update(archived=True)
+        if isinstance(returned_ids, list):
+            for fav_id in list_favs:
+                if fav_id not in returned_ids:
+                    outdated_ids.append(fav_id)
+            if len(outdated_ids) > 0:
+                ProjectedVacancyFavorite.objects.filter(fv_seq_num__in=outdated_ids).update(archived=True)
```
Comment (scott062) [inline talentmap_api/fsbid/services/projected_vacancies.py:276]: Indent (same as above)

**Pull request merged**